cluster.tf

```diff
@@ -0,0 +1,42 @@
+# GKE cluster
+resource "google_container_cluster" "primary" {
+  name     = local.resource_name
+  location = data.google_compute_zones.available.region
+
+  remove_default_node_pool = true
+  initial_node_count       = 1
+  network                  = data.ns_connection.network.outputs.vpc_name
+  subnetwork               = data.ns_connection.network.outputs.public_subnet_names[0]
+}
+
+# Service account to manage permission via IAM
+resource "google_service_account" "default" {
+  account_id   = local.resource_name
+  display_name = "${local.resource_name} Service Account"
+}
+
+# Managed Node Pool
+resource "google_container_node_pool" "primary_nodes" {
+  name     = "${google_container_cluster.primary.name}-node-pool"
+  location = data.google_compute_zones.available.region
+  cluster  = google_container_cluster.primary.name
+
+  node_count = var.node_count
```

Comment:
As above, let's rely on autoscaler to dictate node count

Reply:
Taken care as above